dhis2/dhis2-core
# ci: support running Docker image on arm64 TECH-1454

## To Test

* Run `DHIS2_IMAGE=dhis2/core-dev:2.40-rc-8.5-jre11 docker compose up`. This should pull the necessary images, demo DB and start DHIS2
* `docker inspect dhis2/core-dev:2.40-rc-8.5-jre11 -f "{{ .Architecture }}"` should show arm64 if you are on this architecture


## Details

Building on https://github.com/dhis2/dhis2-core/pull/11981

This PR adds support for building multi-arch Docker images (`linux/amd64` and `linux/arm64`) via Jib and [this profile](https://github.com/dhis2/dhis2-core/pull/12046/files#diff-8a1e9dcacfbca843e23f920dcd5957fbd340b220b140fc56dea6e4d4020ed39bR18-R43), as briefly explained [here](https://github.com/GoogleContainerTools/jib/blob/master/docs/faq.md#how-do-i-specify-a-platform-in-the-manifest-list-or-oci-index-of-a-base-image).


Both the `dev` and the `stable` pipelines will build the Docker images twice:

* The first time the images are tagged with the base tomcat image variant we're running the e2e API tests against (currently `8.5-jre11` and `9.0-jre11`). 
* The second time (only if the tests have passed) the image based on the `9.0-jre11` tomcat variant is built again and tagged with the appropriate branch/tag version (like `latest`, `2.39`, `2.39.0`, etc)

We do this in order to use the convenience of `jib:build` for building multi-arch images, instead of having to tag images separately with their architecture and then combining them in one manifest list.

The `dev` pipeline will use the `core-dev` repo for both of the builds.
The `stable` one will use the `core-dev` repo the first time we build (before the tests) and then once the images have passed the tests - it will build and push to `core`. The reasoning for this is that we shouldn't push untested images or images failing the tests to the main `core` repo.
